tasks/check-data-keys/check-data-keys.yaml

```diff
@@ -24,52 +30,26 @@ spec:
       description: The workspace where the data JSON file resides
   steps:
     - name: check-data-keys
-      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      image: quay.io/konflux-ci/release-service-utils:9089cafbf36bb889b4b73d8c2965613810f13736
       args: ["$(params.systems[*])"]
       script: |
         #!/usr/bin/env bash
         set -ex
 
-        KEYS_JSON='{
-            "releaseNotes": [
-                "product_id",
-                "product_name",
-                "product_version",
-                "product_stream",
-                "cpe",
-                "type",
-                "content.images",
-                "synopsis",
-                "topic",
-                "description",
-                "solution",
-                "references"
-            ],
-            "cdn": [
-                "env"
-            ]
-        }'
-
         if [ ! -f "$(workspaces.data.path)/$(params.dataPath)" ] ; then
             echo "No data JSON was provided."
             exit 1
         fi
 
-        RC=0
-
-        for SYSTEM in "$@" ; do
-            if [[ $(jq -r ".$SYSTEM" <<< "$KEYS_JSON") == null ]] ; then
-                echo "Unsupported system value: $SYSTEM"
-                RC=1
-            fi
+        if ! curl -s --fail-with-body "$(params.schema)" -o /tmp/schema ; then
+            echo "Failed to download schema file: $(params.schema)"
+            exit 1
+        fi
 
-            echo "Checking all required keys exist for: $SYSTEM"
-            for KEY in $(jq -r ".${SYSTEM}[]" <<< "$KEYS_JSON") ; do
-                if [[ $(jq ".$SYSTEM.$KEY" "$(workspaces.data.path)/$(params.dataPath)") == "null" ]] ; then
-                    echo "Missing required $SYSTEM key: $KEY"
-                    RC=1
-                fi
-            done
-        done
+        systemsJSON=$(echo "$@" | jq -R 'split(" ")')
 
-        exit $RC
+        jq --argjson systems "$systemsJSON" '.systems += $systems' \
+            "$(workspaces.data.path)/$(params.dataPath)" > "tmp/systems" 
```

Comment:
I think you should use `/tmp/systems` (missing leading /). I am not sure a tmp dir will exist in the homedir of the container

Reply:
Fixed